kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,287 @@
+use crate::expressions::ColumnName;
+
+use std::collections::HashMap;
+use std::time::Duration;
+
+use serde::de::{self, DeserializeSeed, Deserializer, MapAccess, Visitor};
+use serde::Deserialize;
+
+const SECONDS_PER_MINUTE: u64 = 60;
+const SECONDS_PER_HOUR: u64 = 60 * SECONDS_PER_MINUTE;
+const SECONDS_PER_DAY: u64 = 24 * SECONDS_PER_HOUR;
+const SECONDS_PER_WEEK: u64 = 7 * SECONDS_PER_DAY;
+
+pub(crate) struct StringMapDeserializer<'de> {
```

Comment:
Why do we need this? Can't you just use `MapDeserializer` from serde:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6f967675501f6aa9525d94d74e891eea

Reply:
ah nice thanks for the help nick! this worked but just flagging it requires we consume the Hashmap. i'll try to dig a bit more and see if we can still do `&Hashmap` though probably not a big deal